ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/recovery/WorkerService.java

```diff
@@ -101,12 +107,24 @@ public void doWork (InputStream is, OutputStream os) throws IOException
                 }
             }
 		}
-		out.println("DONE");
+
+		if (RecoveryDriver.isVerboseScan(request)) {
+			for (final RecoveryModule recoveryModule : RecoveryManager.manager().getModules()) {
+				if (recoveryModule instanceof ExtendedRecoveryModule) {
+					if (!((ExtendedRecoveryModule) recoveryModule).isPeriodicWorkSuccessful()) {
+						anyProblems = true;
+					}
+					break;
+				}
+			}
+		}
+
+		out.println(anyProblems ? "ERROR" : "DONE");
 	    }
 	    else
 		out.println("ERROR");
```

Comment:
should not be changed the reported message here? the error is that user does not provided expected parameter `SCAN` or `ASYNC_SCAN` but scanning was not running. Thus reporting `ERROR` could be a bit confusing.

Reply:
No, this is the code that gets reported back to the RecoveryMonitor. The RecoveryMonitor then interprets it and reports the status back to the user.

Reply:
I see, thanks. But the result is similar, isn't it? I mean the `RecoveryMonitor` (https://github.com/jbosstm/narayana/blob/master/ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/tools/RecoveryMonitor.java#L152) reports it as just `ERROR` without satisfactory reasoning, when here it's obviously only an issue of the parameter naming.